plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/conf/__init__.py

```diff
@@ -0,0 +1,213 @@
+
+
+@dataclass
+class RayProviderConf(Dict[str, Any]):
+    type: str = "aws"
+    region: str = "us-west-2"
+    availability_zone: str = "us-west-2a,us-west-2b"
+    cache_stopped_nodes: bool = False
+
+
+@dataclass
+class RayAuthConf(Dict[str, Any]):
+    ssh_user: str = "ubuntu"
+
+
+@dataclass
+class RayNodeConf(Dict[str, Any]):
+    """
+    Additional options in boto docs.
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html
+    """
+
+    InstanceType: str = "m5.large"
+    ImageId: str = "ami-008d8ed4bd7dc2485"
+
+
+@dataclass
+class RsyncConf:
+    """
+    If your code contains more than one modules/files, you need to sync your code to the remote cluster.
+    The code will be synced to a tmp dir on the cluster.
+    code_dir could be a relative dir to where you run your app or a absolute dir.
+    Leave this empty if you don't need to sync code.
+
+    The conf is equivalent to run:
+    rsync {source} {target} --include={include} --exclude={exclude}
+
+    source/target will either be remote/local or local/remote depending on sync_up or sync down
+    Read RayAWSConf for more details.
+    """
+
+    source_dir: Optional[str] = None
+    target_dir: Optional[str] = None
+    include: List[str] = field(default_factory=list)
+    exclude: List[str] = field(default_factory=list)
+
+
+@dataclass
+class RayClusterConf(Dict[str, Any]):
+    """
+    This class maps to a Ray cluster config, e.g:
+    https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml
+    Schema: https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/ray-schema.json
+    """
+
+    # An unique identifier for the head node and workers of this cluster.
+    cluster_name: str = "default"
+
+    # The minimum number of workers nodes to launch in addition to the head
+    # node. This number should be >= 0.
+    min_workers: int = 0
+
+    # The maximum number of workers nodes to launch in addition to the head
+    # node. This takes precedence over min_workers.
+    max_workers: int = 1
+
+    # The initial number of worker nodes to launch in addition to the head
+    # node. When the cluster is first brought up (or when it is refreshed with a
+    # subsequent `ray up`) this number of nodes will be started.
+    initial_workers: int = 0
+
+    autoscaling_mode: RayAutoScalingMode = RayAutoScalingMode.default
+
+    # The autoscaler will scale up the cluster to this target fraction of resource
+    # usage. For example, if a cluster of 10 nodes is 100% busy and
+    # target_utilization is 0.8, it would resize the cluster to 13. This fraction
+    # can be decreased to increase the aggressiveness of upscaling.
+    # This value must be less than 1.0 for scaling to happen.
+    target_utilization_fraction: float = 0.8
+
+    # If a node is idle for this many minutes, it will be removed.
+    idle_timeout_minutes: int = 5
+
+    docker: RayDockerConf = field(default_factory=RayDockerConf)
```

Comment:
This is because RayDockerConf extends Dict. Why is it?

Reply:
1. You did not answer about extending Dict. generally this should be reserved to very special cases. RayDockerConf does not appear to be one.
2. Instead of having special "disabled" indicators in RayDockerConf  in the form of emtpy strings, can you just make it optional here?

```
docker : Optional[RayDockerConf] = None
```
3. Since we are actually supporting Docker here, I think we may need an example demonstrating it.

Reply:
Create #841 to track adding example for docker.

RayDockerConf doesnt need to extend Dict[str, Any] I missed this when cleaning up.